tests/Admin/AdminTest.php

```diff
@@ -632,6 +632,7 @@ public function testSetUniqId(): void
     public function testToString(): void
     {
         $admin = new PostAdmin();
+        $admin->setModelManager($this->createStub(ModelManagerInterface::class));
```

Comment:
this will work now, but not on 5.x when you remove the BC layer.

Reply:
I ll try but when we will remove the bc layer we will chance the doctrine classs utils by the native method get_class from php. So to me it was still working

Reply:
But reading how mongodb wants you to know the real class, you have to use some Metadata class that exists on Persistence bundle and there is a property there.

And they have another deprecated way.

Reply:
Not sure what I should understand herr ^^

Yes it can be done with classmetadata but its not something we shiuld introduce in sonataadmin since it's persistence related.